src/app/admin/invite/admin-users.schema.ts

```diff
@@ -0,0 +1,25 @@
+import { z } from 'zod'
+
+export { zodResolver } from 'mantine-form-zod-resolver'
+
+export const inviteUserSchema = z
+    .object({
+        firstName: z.string().nonempty('is required'),
+        lastName: z.string().nonempty('is required'),
+        email: z.string().nonempty().email('This is not a valid email.'),
+        password: z.string().nonempty('password is required'),
+        isReviewer: z.boolean().optional(),
+        isResearcher: z.boolean().optional(),
+        organizationId: z.string().nonempty('origination must be selected'),
+    })
+    .superRefine((data, ctx) => {
+        if (!data.isReviewer && !data.isResearcher) {
+            ctx.addIssue({
+                path: ['role'],
```

Comment:
i might be blind here but is `role` defined in the schema above? 🤔

Reply:
no it's not,  adding it here is kinda adding some meta info but isn't really typed.  There might be a better way to do this, I'm not sure.

Reply:
yea im still super flaky on the zod stuff too 🤣 it seems theres a million ways to do the same thing